Qc.py

```diff
@@ -23,19 +23,20 @@ def get_average_gpa_best_worst_prof(cur):
                 "FROM("\
                         "SELECT instructor, AVG(grade) "\
                         "FROM Meeting m, Student s "\
-                        "WHERE m.cid = s.cid "\
+                        "WHERE m.cid = s.cid AND s.grade != 'nan' "\
                         "GROUP BY instructor"\
                     ") pag, "\
                     "("\
                         "SELECT MIN(avg), MAX(avg) "\
                         "FROM("\
                                 "SELECT instructor, AVG(grade) "\
                                 "FROM Meeting m, Student s "\
-                                "WHERE m.cid = s.cid "\
+                                "WHERE m.cid = s.cid AND s.grade != 'nan' "\
                                 "GROUP BY instructor"\
                             ") pag"\
                     ") mm "\
-                "WHERE avg = min OR avg = max;")
+                "WHERE avg = min OR avg = max "\
+                "GROUP BY avg, instructor;")
```

Comment:
In question d our grouping results in minimum values being listed before maximum values. Doing the group by operation here is for the sake of consistency.